From b3b458acd392f16eb3f34c052f752866fde1dd1e Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@st.com>
Date: Thu, 12 Dec 2019 11:36:44 +0100
Subject: [PATCH 142/164] drm/panel: otm8009a: allow using non-continuous dsi
 clock

The panel is able to work when dsi clock is non-continuos, thus
the system power consumption can be reduced using such feature.

Add MIPI_DSI_CLOCK_NON_CONTINUOUS to panel's mode_flags.

Upstream-Status: Downstream stm32-gh v5.4-stm32mp [08717bb5b14362b13eec6a1e69dc6d44782c0e2e]
Signed-off-by: Antonio Borneo <antonio.borneo@st.com>
Change-Id: Id58c477c75b659c0a172db62c6cc8207319a7d80
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/153522
Reviewed-by: CITOOLS <smet-aci-reviews@lists.codex.cro.st.com>
Reviewed-by: CIBUILD <smet-aci-builds@lists.codex.cro.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@st.com>
Tested-by: Yannick FERTRE <yannick.fertre@st.com>
---
 drivers/gpu/drm/panel/panel-orisetech-otm8009a.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
index c7b48df8869a..26141a4064be 100644
--- a/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
+++ b/drivers/gpu/drm/panel/panel-orisetech-otm8009a.c
@@ -453,7 +453,7 @@ static int otm8009a_probe(struct mipi_dsi_device *dsi)
 	dsi->lanes = 2;
 	dsi->format = MIPI_DSI_FMT_RGB888;
 	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
-			  MIPI_DSI_MODE_LPM;
+			  MIPI_DSI_MODE_LPM | MIPI_DSI_CLOCK_NON_CONTINUOUS;
 
 	drm_panel_init(&ctx->panel);
 	ctx->panel.dev = dev;
-- 
2.28.0

